apis/k8ssandra/v1alpha1/dseconfig_types.go

```diff
@@ -16,8 +16,2099 @@ limitations under the License.
 
 package v1alpha1
 
-type DSEYaml struct {
+import "k8s.io/apimachinery/pkg/api/resource"
 
-	// TODO
-	Foo string `json:"foo,omitempty"`
+type DseYaml struct {
```

Comment:
To my best knowledge, such a schema, or a validator tool, does not exist. 

The best I could do is modify the existing e2e test to declare almost all the options in dse.yaml. Since the test passes, this means at least that the file generated by config-builder was successfully loaded by DSE (any invalid option in dse.yaml makes DSE fail to start). I also manually verified the resulting dse.yaml file.

The large number of options also begs the question of maintainability. It's already very hard to maintain `CassandraYaml`, we currently use a spreadsheet to help track new fields. We would need to do something equivalent for DseYaml. Especially if/when we will have to support DSE 7.0+.